testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/userdata/ProfileSettingsActivity.kt

```diff
@@ -0,0 +1,119 @@
+package com.rakuten.tech.mobile.testapp.ui.userdata
+
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.Menu
+import android.view.MenuItem
+import android.widget.ImageView
+import com.bumptech.glide.Glide
+import com.bumptech.glide.request.RequestOptions
+import com.rakuten.tech.mobile.miniapp.testapp.R
+import com.rakuten.tech.mobile.testapp.helper.clearWhiteSpaces
+import com.rakuten.tech.mobile.testapp.helper.isInputEmpty
+import com.rakuten.tech.mobile.testapp.ui.base.BaseActivity
+import com.rakuten.tech.mobile.testapp.ui.settings.AppSettings
+import kotlinx.android.synthetic.main.profile_settings_activity.*
+
+class ProfileSettingsActivity : BaseActivity() {
+
+    private lateinit var settings: AppSettings
+    private lateinit var profileUrl: String
+
+    private val nameTextWatcher = object : TextWatcher {
+        override fun afterTextChanged(s: Editable?) {}
+
+        override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+
+        @SuppressLint("SyntheticAccessor")
+        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+            validateNameInput()
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        settings = AppSettings.instance
+        profileUrl = settings.profilePictureUrl
+        showBackIcon()
+        setContentView(R.layout.profile_settings_activity)
+        renderProfileSettingsScreen()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.settings_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                finish()
+                return true
+            }
+            R.id.settings_menu_save -> {
+                updateProfile(editProfileName.text.toString())
+                return true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun renderProfileSettingsScreen() {
+        setProfileImage(Uri.parse(settings.profilePictureUrl))
+        editProfileName.setText(settings.profileName)
+        editProfileName.addTextChangedListener(nameTextWatcher)
+        textEditPhoto.setOnClickListener { openGallery() }
```

Comment:
Tap on photo `ImageView` should call `openGallery()` too.

Reply:
Yeah, thinking about the UX e.g. Tap or Long Tap. For now, that's could be optional as there is already a primary view called "Edit Photo".